test/e2e/specs/site-editor/style-variations.spec.js

```diff
@@ -0,0 +1,299 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.use( {
+	siteEditorStyleVariations: async (
+		{ admin, editor, page, pageUtils, requestUtils },
+		use
+	) => {
+		await use(
+			new SiteEditorStyleVariations( {
+				admin,
+				editor,
+				page,
+				pageUtils,
+				requestUtils,
+			} )
+		);
+	},
+} );
+
+test.describe( 'Global styles variations', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme(
+			'gutenberg-test-themes/style-variations'
+		);
+		await requestUtils.deleteAllTemplates( 'wp_template' );
+		await requestUtils.deleteAllTemplates( 'wp_template_part' );
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await Promise.all( [
+			requestUtils.deleteAllTemplates( 'wp_template' ),
+			requestUtils.deleteAllTemplates( 'wp_template_part' ),
+		] );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.activateTheme( 'twentytwentyone' );
+	} );
+
+	test( 'should have three variations available with the first one being active', async ( {
+		admin,
+		editor,
+		siteEditorStyleVariations,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await editor.toggleGlobalStyles();
+		await siteEditorStyleVariations.openOtherStyles();
+
+		const variations = await siteEditorStyleVariations.getAvailableStyleVariations();
+
+		expect( await variations.count() ).toEqual( 3 );
+		expect( await variations.first().getAttribute( 'class' ) ).toContain(
+			'is-active'
+		);
+		expect(
+			await variations.nth( 1 ).getAttribute( 'class' )
+		).not.toContain( 'is-active' );
+		expect(
+			await variations.nth( 2 ).getAttribute( 'class' )
+		).not.toContain( 'is-active' );
```

Comment:
I think it's a good opportunity for us to rethink the assertions here. Generally, we don't want to assert class names since they are often just implementation details. Instead, is there any accessible attribute we could get here to represent _active_? Like `aria-*` attribute or even `isVisible()` check.

If that's not possible for now, then I think we can safely say that the app is also not accessible to screen readers and we should fix that in a future PR. In that case, we can continue asserting class names here but we should add some comments above explaining why.

Reply:
Good advice, thanks. I'll look at this next.

Reply:
I'm taking care of adding aria attributes over in https://github.com/WordPress/gutenberg/pull/41591

I probably would have done it here, but I found a bug as well where the active state cannot be accessed for some theme.json files.

So I think we can leave this until https://github.com/WordPress/gutenberg/pull/41591 is merged. I'll add a comment to that effect here.